### ISSUE TITLE
"Save Image As" JPG option ignored on batch upscale

### ISSUE THREAD

**Issue by nilei** [upscayl/upscayl#620]
### Checklist

- [X] I have checked that this issue isn't addressed in [the FAQ](https://github.com/upscayl/upscayl#-faq).
- [X] I have checked that this issue isn't addressed in any other issue or pull request.
- [ ] I have tried [setting my GPU ID](https://github.com/upscayl/upscayl/wiki/Guide#gpu-id).
- [ ] I am using Windows and I've tried [common troubleshooting steps](https://github.com/upscayl/upscayl/wiki/Troubleshooting).

### Describe the Bug

If a batch upscale is performed and the image format is set to JPG, the new images are saved as PNG.

### To Reproduce

Set JPG as image format.
Perform batch upscaling.
Check the image format of the new images.

### Version (or commit hash)

2.9.5

### Select your Operating System

MacOS

### OS Version

Ventura 13.4.1

### GPU Name

M1

### Expected Behavior

The new images should all be saved in JPG format.

### Paste the logs

```sh
⚙️ Getting saveImageAs from localStorage:  png
⚙️ Getting model from localStorage:  {"label":"Digital Art","value":"realesrgan-x4plus-anime"}
⚙️ Setting gpuId to empty string
🔀 Setting model to 
🐞 BACKEND REPORTED:  💾 Updating Save Output Folder:  false
🔀 Setting model to realesrgan-x4plus-anime
🐞 BACKEND REPORTED:  📐 Updating Compression:  0
🐞 BACKEND REPORTED:  📝 Updating Overwrite:  false
⚙️ Getting saveImageAs from localStorage:  png
⚙️ Getting model from localStorage:  {"label":"Digital Art","value":"realesrgan-x4plus-anime"}
⚙️ Setting gpuId to empty string
⚙️ Getting rememberOutputFolder from localStorage:  false
⚙️ Getting saveImageAs from localStorage:  jpg
⚙️ Getting model from localStorage:  {"label":"Digital Art","value":"realesrgan-x4plus-anime"}
⚙️ Setting gpuId to empty string
🔀 Setting model to 
🔀 Setting model to realesrgan-x4plus-anime
🔄 Resetting image paths
🔀 Model changed:  realesrgan-x4plus-anime
🔀 Setting model to realesrgan-x4plus-anime
🔄 Resetting Upscaled Image Path
🏁 FOLDER_UPSCAYL
🐞 BACKEND REPORTED:  🖼️ Updating No Image Processing:  false
🐞 BACKEND REPORTED:  📐 Updating Compression:  0
🐞 BACKEND REPORTED:  🗑️ Deleting .DS_Store file
🐞 BACKEND REPORTED:  🛑 Updating Stopped:  false
🚧 FOLDER_UPSCAYL_PROGRESS:  [0 Apple M1 Pro]  queueC=0[1]  queueG=0[1]  queueT=0[1]
[0 Apple M1 Pro]  bugsbn1=0  bugbilz=50  bugcopc=0  bugihfa=0
[0 Apple M1 Pro]  fp16-p/s/a=1/1/1  int8-p/s/a=1/1/1
[0 Apple M1 Pro]  subgroup=32  basic=1  vote=1  ballot=1  shuffle=1
```

**Comment by aaronliu0130** [comment]
Please paste the full logs.

**Comment by aaronliu0130** [comment]
did you stop the upscayl while in progress?
could you show the logs?

**Comment by Plugway** [comment]
I discovered this too. I don't think this is a bug, it's more of a feature. The program initially saves all images as png, and after upscaling it downscales them if needed and converts to jpg.
It would be nice if all images were initially saved as jpg to reduce disk usage and free disk space requirements, since png takes up tens of times more space than jpg.

My os is Windows, but I don't think that matters.

**Comment by NayamAmarshe** [comment]
Yes that is correct. Only PNGs are saved first and then are converted to JPGs.

I'll make them save as JPG by default in the next version.

**Comment by aaronliu0130** [comment]
If so, could you make it output PNGs by default for digital art?

**Comment by Talk2Giuseppe** [comment]
This isn't making sense.  If you select the jpg images option, and run a batch on 100 images, you get png - that's  a feature?!  Am I missing something?  How in the world do we get a BATCH of images in jpg format then?  Do it one at a time?  If so, what's the point of batch mode?  PNG is useless at ultra high levels of resolution... Well, maybe not useless, but it is more cumbersome on a batch of images versus jpg.

Sorry, I'm just a week behind on a project and this was the first upscaler to have done a decent job on the images.  I was having high hopes to finally get the project done.  But looks like I'll need to wait or find something else to upscale this batch of 150 images.

At least the issue is fixable.  Thanks!

**Comment by NayamAmarshe** [comment]
> This isn't making sense.  If you select the jpg images option, and run a batch on 100 images, you get png - that's  a feature?!  Am I missing something?  How in the world do we get a BATCH of images in jpg format then?  Do it one at a time?  If so, what's the point of batch mode?  PNG is useless at ultra high levels of resolution... Well, maybe not useless, but it is more cumbersome on a batch of images versus jpg.
> 
> 
> 
> Sorry, I'm just a week behind on a project and this was the first upscaler to have done a decent job on the images.  I was having high hopes to finally get the project done.  But looks like I'll need to wait or find something else to upscale this batch of 150 images.
> 
> 
> 
> At least the issue is fixable.  Thanks!

You need to wait for the batch process to finish to get the JPGs. If even after the batch process is done and you don't see any JPGs,you can try this tool for the time being to convert PNG to JPG: https://www.shutterencoder.com/en/

**Comment by Talk2Giuseppe** [comment]
Some additional testing reveals that if the starting image is <image.jpeg>, it will not convert to <image.jpg> at it's conclusion of the batch.  So if you start with a file with a .jpeg ext, just rename the ext to .jpg and run batch again.  It will work as expected.

Also, I seem to remember it would rename the files to include the upsale type and what upscale multiplier - but that doesn't seem to be working.  It's also possible Upscayl never did that and I am thinking of another upscaler application.  In any event, it would be nice if the final filename did have some reflection of what it is.

**Comment by aaronliu0130** [comment]
Hmm, the .jpeg behavior is weird and probably something we should fix.
Upscayl changes the name in individual-file upscayling mode. In batch mode, it simply renames the folder.

**Comment by Talk2Giuseppe** [comment]
Ahhhhh... Ok, yes - that makes sense.  I tend to play with this late into the night.  Things start getting pretty foggy at that point.

The folder rename is great - but the file rename is better!  Especially if we're putting images through various processes over the course of several days.  As a team effort, we generally will ask for "the file at 5620 x 5620".  Once it's pulled from the folder, it's hard to search for it.

Thanks for the quick response!

**Comment by NayamAmarshe** [comment]
> Hmm, the .jpeg behavior is weird and probably something we should fix. Upscayl changes the name in individual-file upscayling mode. In batch mode, it simply renames the folder.

Can't reproduce the JPEG issue.
<img width="347" alt="image" src="https://github.com/upscayl/upscayl/assets/25067102/8262000e-83f3-458e-a2f6-3c9694187a5e">

**Comment by nilei** [comment]
The problem is not the end result, but the workflow PNG => JPG. In a batch process, PNG files are obviously created first, which are then converted to JPG files at the end. During the upscaling process, however, the PNG files take up a lot more space. For example, with 1,000 JPG files of approx. 100KB each, too many PNG files are simply created before they are then changed to JPG. I choose the JPG option to reduce the storage requirements. But that doesn't do me much good if the space is still needed during the upscaling process.

**Comment by NayamAmarshe** [comment]
> The problem is not the end result, but the workflow PNG => JPG. In a batch process, PNG files are obviously created first, which are then converted to JPG files at the end. During the upscaling process, however, the PNG files take up a lot more space. For example, with 1,000 JPG files of approx. 100KB each, too many PNG files are simply created before they are then changed to JPG. I choose the JPG option to reduce the storage requirements. But that doesn't do me much good if the space is still needed during the upscaling process.

Got it, I've fixed this in 2.9.6. Also added a WEBP option.

**Comment by aaronliu0130** [comment]
(reopen if needed)
